internal/providers/terraform/azure/storage_account.go

```diff
@@ -0,0 +1,195 @@
+		return nil
+	}
+
+	accountReplicationType := d.Get("account_replication_type").String()
+	accountTier := d.Get("account_tier").String()
+	accessTier := "Hot"
+	if d.Get("access_tier").Type != gjson.Null {
+		accessTier = d.Get("access_tier").String()
+	}
+
+	productName := map[string]string{
+		"Standard": "Blob Storage",
+		"Premium":  "Premium Block Blob",
+	}[accountTier]
+
+	if productName == "" {
+		log.Warnf("Unrecognized account tier for resource %s: %s", d.Address, accountTier)
+		return nil
+	}
+
+	validPremiumReplicationTypes := []string{"ZRS", "LRS"}
+	validStandardReplicationTypes := []string{"LRS", "GRS", "RAGRS"}
+
+	if accessTier == "Premium" {
+		if !Contains(validPremiumReplicationTypes, accountReplicationType) || !Contains(validStandardReplicationTypes, accountReplicationType) {
+			log.Warnf("%s redundancy does not supports for %s performance tier", accountReplicationType, accountTier)
+		}
+	}
+
+	var capacity, writeOperations, listOperations, readOperations, otherOperations, dataRetrieval, dataWrite, blobIndex *decimal.Decimal
+
+	if accountReplicationType == "RAGRS" {
+		accountReplicationType = "RA-GRS"
+	}
+
+	skuName := fmt.Sprintf("%s %s", accessTier, accountReplicationType)
+	if accountTier == "Premium" {
+		skuName = fmt.Sprintf("%s %s", accountTier, accountReplicationType)
+	}
+
+	if u != nil && u.Get("storage_gb").Exists() {
+		capacity = decimalPtr(decimal.NewFromInt(u.Get("storage_gb").Int()))
+
+		if accessTier == "Hot" {
+			dataStorageTiers := []int{51200, 512000}
+			dataStorageQuantities := usage.CalculateTierBuckets(*capacity, dataStorageTiers)
+
+			costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity (first 50TB)", skuName, "0", productName, &dataStorageQuantities[0]))
+			if dataStorageQuantities[1].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity (next 450TB)", skuName, "51200", productName, &dataStorageQuantities[1]))
+			}
+			if dataStorageQuantities[2].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity (over 500TB)", skuName, "512000", productName, &dataStorageQuantities[2]))
+			}
+		} else {
+			costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity", skuName, "0", productName, capacity))
+		}
+	} else {
+		var unknown *decimal.Decimal
+
+		costComponents = append(costComponents, blobDataStorageCostComponent(location, "Capacity", skuName, "0", productName, unknown))
+	}
+
+	if u != nil && u.Get("monthly_write_operations").Exists() {
+		writeOperations = decimalPtr(decimal.NewFromInt(u.Get("monthly_write_operations").Int()))
+	}
+	meterName := "/Write Operations$/"
+	costComponents = append(costComponents, blobOperationsCostComponent(location, "Write operations", "10K operations", skuName, meterName, productName, writeOperations, 10000))
```

Comment:
```suggestion
	costComponents = append(costComponents, blobOperationsCostComponent(
		location,
		"Write operations",
		"10K operations",
		skuName,
		"/Write Operations$/",
		productName,
		writeOperations,
		10000,
	))
```
This doesn't need to change. I think breaking this into lines makes it easier to read and doesn't need a redundant `meterName` var. But it doesn't affect the code, it's just a personal stylistic thing, so will leave it up to you.

Reply:
Yes, u are right - it looks better :) 
+ I noticed and fixed one nuance here, I don't know why, but Hot RA-GRS List and Create Container Operations component does not exists